storm-core/src/clj/backtype/storm/daemon/executor.clj

```diff
@@ -570,7 +575,11 @@
                                                          (tasks-fn out-stream-id values))
                                              rooted? (and message-id has-ackers?)
                                              root-id (if rooted? (MessageId/generateId rand))
-                                             out-ids (fast-list-for [t out-tasks] (if rooted? (MessageId/generateId rand)))]
+                                             out-ids (fast-list-for [t out-tasks] (if rooted? (MessageId/generateId rand)))
+                                             sampler? (sampler)]
+                                         (when sampler?
+                                           (.notify rate-tracker (count out-tasks))
```

Comment:
Wait I just thought about this again.  How in the world can we get an accurate rate if we are only counting a subset of the tuples?  Wouldn't we also have to increase the rate proportionally to the sampling rate?